src/beast/beast/unit_test/suite.h

```diff
@@ -346,11 +346,13 @@ bool
 suite::expect (Condition shouldBeTrue,
     String const& reason)
 {
+    bool const b =
+        static_cast<bool>(shouldBeTrue);
     if (shouldBeTrue)
```

Comment:
`if(shouldBeTrue)` will fail to compile if `Condition` is `std::shared_ptr<T>`. My motivation for making this change is so that you don't have to write `expect( ! ! sle)` when you want to assure that a `shared_ptr` is not null.